e2e_tests/src/e2e_conduit/runner.py

```diff
@@ -0,0 +1,103 @@
+import atexit
+import os
+import logging
+import shutil
+import subprocess
+import sys
+import tempfile
+import time
+
+import yaml
+
+from e2e_common.util import find_binary
+from e2e_conduit.subslurp import subslurp
+
+logger = logging.getLogger(__name__)
+
+
+class ConduitE2ETestRunner:
+    def __init__(self, conduit_bin, keep_temps=False):
+        self.conduit_bin = find_binary(conduit_bin, binary_name="conduit")
+        self.keep_temps = keep_temps
+
+    def setup_scenario(self, scenario):
+
+        # Setup conduit_dir for conduit data dir
+        scenario.conduit_dir = tempfile.mkdtemp()
+        if not self.keep_temps:
+            atexit.register(shutil.rmtree, scenario.conduit_dir, onerror=logger.error)
+        else:
+            logger.info("leaving temp dir %r", scenario.conduit_dir)
+
+        scenario.accumulated_config = {
+            "conduit_dir": scenario.conduit_dir,
+        }
+
+        for plugin in [scenario.importer, *scenario.processors, scenario.exporter]:
+            plugin.setup(scenario.accumulated_config)
+            plugin.resolve_config_input()
+            plugin.resolve_config_output()
```

Comment:
not even a nit: is this the expected usage pattern? If so, maybe it's worth having the base case implement a new method `resolve_config()` which would be called instead.